supervision/metrics/mean_average_precision.py

```diff
@@ -23,6 +27,27 @@
 
 
 class MeanAveragePrecision(Metric):
+    """
+    Mean Average Precision (mAP) is a metric used to evaluate object detection models.
+    It is the average of the precision-recall curves at different IoU thresholds.
+
+    Example:
+        ```python
+        import supervision as sv
+        from supervision.metrics import MeanAveragePrecision
+
+        predictions = sv.Detections(...)
+        targets = sv.Detections(...)
+
+        map_metric = MeanAveragePrecision()
+        map_result = map_metric.update(predictions, targets).compute()
+
+        print(map_result)
```

Comment:
Until now, in our examples, I've tried to show the format of the obtained result, for example

```
print(confusion_matrix.matrix)
# np.array([
#    [0., 0., 0., 0.],
#    [0., 1., 0., 1.],
#    [0., 1., 1., 0.],
#    [1., 1., 0., 0.]
# ])
```

From what I see, you've stopped doing that for metrics. I think we should continue doing it.